### ISSUE TITLE
Delegation errors with forward/reverse domain and slave root (.) zone (RFC 7706)

### ISSUE THREAD

**Issue by tokred** [PowerDNS/pdns#8716]
- Program: Authoritative 
 - Issue type: Bug report

### Short description

After using Knot and NSD for quite a while, I am testing to build an internal NS using PowerDNS. I have successfully set up a non-signed and internal-only domain `unimatrix.zero` + related `10.in-addr.arpa` reverse zone, all working fine.
As soon as I add the root zone `.` as slave (according to RFC 7706), I cannot edit my main zone anymore because of `[Error] No delegation for zone 'unimatrix.zero' in parent '.'` (of course not?) and `pdnsutil check-all-zones` complains about errors in the `.` zone.

### Environment

 - Operating system: Arch Linux
 - Software version: PowerDNS 4.2.1, SQLite3 backend
 - Software source: OS repository 

### Steps to reproduce

1. Created one main native zone and an in-addr.arpa reverse zone, and populated with records, no DNSSEC.
2. Enabled `slave=yes` and DNSSEC for SQLite backend with `gsqlite3-dnssec=yes`.
3. Created slave zone `.` with some root servers as masters via `pdnsutil create-slave-zone . 192.0.47.132 192.228.79.201`, AXFR works fine, PRESIGNED is automatically set to 1.
4. From now on, `pdnsutil edit-zone unimatrix.zero` throws delegation error, even without any zone change.

### Expected behaviour

I would expect that mirroring the presigned root zone would not affect my own zones, and I would expect that `pdnsutil check-all-zones` would not complain about the `.` zone at all. Identical setup worked fine with Knot.

### Actual behaviour

`pdnsutil check-all-zones` throws the error for my zones `[Error] No delegation for zone 'unimatrix.zero' in parent '.'`, and additional errors for the root zone, for each and every TLD:
```
...
[Warning] 'com|RRSIG' in zone '.' is occluded by a delegation at 'com'
...
[Error] Following record is auth=1, run pdnsutil rectify-zone?: com IN RRSIG NSEC 8 1 86400 20200201050000 20200119040000 33853 . ja9dZ...
...
```
Running `pdnsutil rectify-all-zones` did not change anything.

### Other information

```
$ pdnsutil show-zone unimatrix.zero
This is a Native zone
Zone is not actively secured
Metadata items: None
No keys for zone 'unimatrix.zero'.

$ pdnsutil list-zone unimatrix.zero
$ORIGIN .
authns1.unimatrix.zero	3600	IN	A	10.1.1.153
unimatrix.zero	3600	IN	NS	authns1.unimatrix.zero.
unimatrix.zero	3600	IN	SOA	authns1.unimatrix.zero hostmaster.unimatrix.zero 2020011603 14400 3600 1209600 600

$ pdnsutil show-zone .
This is a Slave zone
Masters: 192.0.47.132:53 192.228.79.201:53 [2001:500:2f::f]:53 [2620:0:2d0:202::132]:53 192.33.4.12:53 
Last time we got update from master: Sun 2020-01-19 11:36:30
SOA serial in database: 2020011900
Refresh interval: 1800 seconds
Metadata items: 
	PRESIGNED	1
Zone is presigned
Zone has NSEC semantics
keys: 
KSK, tag = 20326, algo = 8, bits = 2048
DNSKEY = . IN DNSKEY 257 3 8 AwEAAaz/tAm8yTn4Mfeh5eyI96WSVexTBAvkMgJzkKTOiW1vkIbzxeF3+/4RgWOq7HrxRixHlFlExOLAJr5emLvN7SWXgnLh4+B5xQlNVz8Og8kvArMtNROxVQuCaSnIDdD5LKyWbRd2n9WGe2R8PzgCmr3EgVLrjyBxWezF0jLHwVN8efS3rCj/EWgvIWgb9tarpVUDK/b58Da+sqqls3eNbuv7pr+eoZG+SrDK6nWeL3c6H5Apxz7LjVc1uTIdsIXxuOLYA4/ilBmSVIzuDWfdRUfhHdY6+cn8HFRm+2hM8AnXGXws9555KrUB5qihylGa8subX2Nn6UwNR1AkUTV74bU=; ( RSASHA256 ) 
DS = . IN DS 20326 8 1 ae1ea5b974d4c858b740bd03e3ced7ebfcbd1724 ; ( SHA1 digest )
DS = . IN DS 20326 8 2 e06d44b80b8f1d39a95c0b0d7c65d08458e880409bbc683457104237c7f8ec8d ; ( SHA256 digest )
DS = . IN DS 20326 8 4 538f47ba9bb88908e1dc335d6dfd51ca66b4d824192e6e6e210ae8cc18ece46a0f62b9f0d2f88dfc87d4bb8b8aed21cb ; ( SHA-384 digest )
ZSK, tag = 33853, algo = 8, bits = 2048
DNSKEY = . IN DNSKEY 256 3 8 AwEAAeN+h0loXPKt7lFdW2zKIDkVHyJ1aYGUVE1dMNBlRH3kTn40JKcHiPOs+fy0OFVCBwoKa1s9qZtdyP1UC0hgKoldj3oELK1yLI5MUbTMcNkWbBMRuxRz/CgZJu3IxcmuZWZMbn4LQDMj5YeiUiuWns5vipFGWWpyPyozQXmenSWOK2GJOwcm7I/DyHVtVdztTvqiHqzy2aRoxwPhmEuAoYzzuNJJw6JNEnXaN/7l2TIciskFyPVPBFZYHnk+1ma906dfehIR190z3lh1ZESL2Yy3VIE2QGpRU6Px4ydH5sXxZ2wSMgqNNga4kjnfM1msBqk3EI48RvTTkuV0yb1eFuU=; ( RSASHA256 ) 
DS = . IN DS 33853 8 1 b2ed05b05041dbbcff4b2c69db7edd9884afa92c ; ( SHA1 digest )
DS = . IN DS 33853 8 2 322405011ce973ad545d0c5334f1fc6f8b5f4209a424e400a6954a34379defcc ; ( SHA256 digest )
DS = . IN DS 33853 8 4 b81edb7c8c393ccc18c1eaee841beb99aabe740c95724dbcab2fd8cc5c6b84199421cb1589267a81333c3ae617d0d8fe ; ( SHA-384 digest )
```

Thanks for looking into this! Regards

**Comment by pieterlexis** [comment]
>  As soon as I add the root zone . as slave (according to RFC 7706), I cannot edit my main zone anymore because of [Error] No delegation for zone 'unimatrix.zero' in parent '.' (of course not?) and pdnsutil check-all-zones complains about errors in the . zone.

I am surprised this happens, how do you edit the zone?

Regarding the check-zone error for the root zone, perhaps PowerDNS shouldn't do check-zone on presigned zones?

**Comment by tokred** [comment]
> I am surprised this happens, how do you edit the zone?

Sorry, forgot to mention, with `pdnsutil edit-zone`.

**Comment by Habbie** [comment]
That delegation check was not ready for your fragmented view on reality. So that's a bug.

check-zone's terrible handling of presigned zones is a different issue, and it should probably warn that results can be confusing/wrong for such zones.

**Comment by aba-rechsteiner** [comment]
if i understand correctly this is "only" a bug in the pdnsutil, but the pdns works as expected?

**Comment by Habbie** [comment]
> if i understand correctly this is "only" a bug in the pdnsutil, but the pdns works as expected?

As far as I know, yes!

**Comment by tokred** [comment]
I would agree, from what I saw pdns worked fine but only pdnsutil raised the errors. In the meantime, I have implemented the RFC 7706 approach on the resolver instead of the authoritative name server.

Regards

**Comment by aba-rechsteiner** [comment]
> In the meantime, I have implemented the RFC 7706 approach on the resolver instead of the authoritative name server.

interesting, how did you do that?

**Comment by tokred** [comment]
@aba-rechsteiner, RF8806 (which obsoleted 7706) [Annex B](https://datatracker.ietf.org/doc/html/rfc8806#appendix-B) contains example configurations for resolvers. I like Unbound's approach because it allows using the root zone data as local cache only, with a fallback option to query the actual root servers in case of errors.
With my previous setup with pdns serving the slave root zone I had some rare issues after booting because of (I think) a strange interdependency of resolver and pdns, leading to some timeouts.

Regards
